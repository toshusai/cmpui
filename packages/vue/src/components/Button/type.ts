export const ButtonVariant = {
  primary: "primary",
  secondary: "secondary",
} as const;

export type ButtonVariant = (typeof ButtonVariant)[keyof typeof ButtonVariant];

export const ButtonSize = {
  S: "S",
  M: "M",
  L: "L",
} as const;

export type ButtonSize = (typeof ButtonSize)[keyof typeof ButtonSize];

export type ButtonProps = {
  variant: ButtonVariant;
  size: ButtonSize;
  disabled: boolean;
};
