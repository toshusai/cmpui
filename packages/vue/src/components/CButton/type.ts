export const CButtonVariant = {
  primary: "primary",
  secondary: "secondary",
} as const;

export type CButtonVariant =
  (typeof CButtonVariant)[keyof typeof CButtonVariant];

export const CButtonSize = {
  S: "S",
  M: "M",
  L: "L",
} as const;

export type CButtonSize = (typeof CButtonSize)[keyof typeof CButtonSize];

export type CButtonProps = {
  variant?: CButtonVariant;
  size?: CButtonSize;
  disabled?: boolean;
};
