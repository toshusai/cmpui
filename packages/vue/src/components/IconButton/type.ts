export const IconButtonSize = {
  S: "S",
  M: "M",
} as const;

export type IconButtonSize =
  (typeof IconButtonSize)[keyof typeof IconButtonSize];

export type IconButtonProps = {
  bordered?: boolean;
  disabled?: boolean;
  rounded?: boolean;
  selected?: boolean;
  size?: IconButtonSize;
};
