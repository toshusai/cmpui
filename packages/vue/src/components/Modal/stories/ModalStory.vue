<script setup lang="ts">
import { ref } from "vue";
import Button from "../../Button/Button.vue";
import Modal from "../Modal.vue";

const show = ref(false);
</script>

<template>
  <Modal :show="show" title="Confirm" @close="show = false">
    <div style="padding: 8px">
      <p style="font-size: 14px; margin: 0">
        Are you sure you want to delete this item?
      </p>
      <div
        style="display: flex; gap: 8px; justify-content: right; margin-top: 8px"
      >
        <Button size="S" variant="secondary" @click="show = false">
          Cancel
        </Button>
        <Button size="S" @click="show = false">OK</Button>
      </div>
    </div>
  </Modal>
  <Button @click="show = true">Open</Button>
</template>
