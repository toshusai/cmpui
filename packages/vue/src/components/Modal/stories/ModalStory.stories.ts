import type { Meta, StoryObj } from "@storybook/vue3";
import ModalStory from "./ModalStory.vue";

const meta = {
  title: "components/Modal",
  component: ModalStory,
  tags: ["autodocs"],
} satisfies Meta<typeof ModalStory>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {};
