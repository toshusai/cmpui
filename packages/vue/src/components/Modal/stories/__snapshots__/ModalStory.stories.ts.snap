// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Modal Default smoke-test 1`] = `
<button class="cmpui_button"
        data-variant="primary"
        data-size="M"
>
  Open
</button>
`;
